Automattic/jetpack
# Settings UI: fix alert about unsaved settings showing twice

#### Changes proposed in this Pull Request:

* don't alert user twice about unsaved settings
* remove a lot of unused code to fire warnings for unsaved module options since now everything is a setting
* remove unnecessary usage of `bindActionCreators`

#### Testing instructions:

* make changes and don't save. Switch tab, it should show the alert. Choose OK to switch. Switch tab again, it should not show the alert now.
* make sure initial state is ok


#### Proposed changelog entry for your changes:
